src/components/TodoistQuickAdd.tsx

```diff
@@ -0,0 +1,134 @@
+import { useState, useEffect, useRef } from 'react'
+
+const LOADING_TIMEOUT = 5000 // 5sec
+const SCRIPT_ELM_ID = 'todoist-quick-add'
+const TODOIST_HOST = 'todoist.com'
+
+type ShowProps = {
+    content: string
+    todoistHost: string
+    priority?: number
+    date?: string
+    theme?: number
+}
+
+declare global {
+    interface Window {
+        TodoistQuickAddSDK: {
+            show: (parms: ShowProps) => void
+        }
+    }
+}
+
+type Props = {
+    content: string
+    priority?: number
+    date?: string
+    theme?: number
+    todoistHost?: string
+    onAdd?: () => void
+    onClose?: () => void
+    onLoadingSuccess?: () => void
+    onLoadingError?: () => void
+}
+
+type CustomMessageEvent = {
+    data: string
+}
+
+type MessageJSON = {
+    type: string
+}
+
+function TodoistQuickAdd({
+    content,
+    priority,
+    date,
+    theme,
+    todoistHost,
+    onAdd,
+    onClose,
+    onLoadingSuccess,
+    onLoadingError,
+}: Props): null {
+    const [isLoaded, setIsLoaded] = useState<boolean>(false)
+    const loadingTimerToClear = useRef<number | undefined>(undefined)
+
+    // Set up post message listener
+    useEffect(() => {
+        function handleMessageReceived(event: CustomMessageEvent) {
+            const data = event.data
```

Comment:
I'm no expert at this, but I believe for security reasons it is important that `origin` of message is verified.

Reply:
Good point. I'll make the origin check explicit.